### ISSUE TITLE
MTase-linker install and check_installation are broken in nanomotif 0.4.16

### ISSUE THREAD

**Issue by rattei** [MicrobialDarkMatter/nanomotif#83]
Hello,

in nanomotif 0.4.16 it is no longer possible to run "nanomotif MTase-linker install". Also "nanomotif check_installation" is broken.

```
[foo@bar nanomotif]$ nanomotif --version
nanomotif 0.4.16


[foo@bar nanomotif]$ nanomotif check_installation
100% (4 of 4) |#######################################################################################################################################################################################| Elapsed Time: 0:00:05 Time:  0:00:05
100% (2 of 2) |#######################################################################################################################################################################################| Elapsed Time: 0:00:02 Time:  0:00:02
Traceback (most recent call last):
  File "/lisc/app/conda/miniforge3/envs/nanomotif-0.4.16/bin/nanomotif", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/lisc/app/conda/miniforge3/envs/nanomotif-0.4.16/lib/python3.12/site-packages/nanomotif/main.py", line 620, in main
    check_install(args, pl)
  File "/lisc/app/conda/miniforge3/envs/nanomotif-0.4.16/lib/python3.12/site-packages/nanomotif/main.py", line 432, in check_install
    shutil.rmtree(args.out)
  File "/lisc/app/conda/miniforge3/envs/nanomotif-0.4.16/lib/python3.12/shutil.py", line 759, in rmtree
    _rmtree_safe_fd(stack, onexc)
  File "/lisc/app/conda/miniforge3/envs/nanomotif-0.4.16/lib/python3.12/shutil.py", line 703, in _rmtree_safe_fd
    onexc(func, path, err)
  File "/lisc/app/conda/miniforge3/envs/nanomotif-0.4.16/lib/python3.12/shutil.py", line 662, in _rmtree_safe_fd
    os.rmdir(name, dir_fd=dirfd)
OSError: [Errno 39] Directory not empty: 'nanomotif_install_check/logs'


[foo@bar nanomotif]$ nanomotif MTase-linker install 
Traceback (most recent call last):
  File "/lisc/app/conda/miniforge3/envs/nanomotif-0.4.16/bin/nanomotif", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/lisc/app/conda/miniforge3/envs/nanomotif-0.4.16/lib/python3.12/site-packages/nanomotif/main.py", line 591, in main
    os.environ["POLARS_MAX_THREADS"] =str(args.threads)
                                          ^^^^^^^^^^^^
AttributeError: 'Namespace' object has no attribute 'threads'
```

best,
Thomas

**Comment by janeacook** [comment]
Hello,

I am having similar issues as @rattei. I have tried multiple combinations of python 3.12 and 3.9 with nanomotif 0.4.16 and 0.4.9 (the version I had been using, but decided to update to use the MTase-linker module). I have different check_installation errors, which I have pasted below:

`$ nanomotif check_installation
Traceback (most recent call last):
  File "/oak/stanford/scg/lab_asbhatt/cookja/software/micromamba/envs/nanomotif_416/bin/nanomotif", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/oak/stanford/scg/lab_asbhatt/cookja/software/micromamba/envs/nanomotif_416/lib/python3.12/site-packages/nanomotif/main.py", line 620, in main
    check_install(args, pl)
  File "/oak/stanford/scg/lab_asbhatt/cookja/software/micromamba/envs/nanomotif_416/lib/python3.12/site-packages/nanomotif/main.py", line 420, in check_install
    motifs = find_motifs(args, pl, pileup=pileup, assembly=assembly)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/oak/stanford/scg/lab_asbhatt/cookja/software/micromamba/envs/nanomotif_416/lib/python3.12/site-packages/nanomotif/main.py", line 79, in find_motifs
    contig_mods_to_keep, contig_mods_to_remove = nm.dataload.extract_contig_mods_with_sufficient_information(pileup, assembly, min_mods_pr_contig, min_mod_frequency)
                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/oak/stanford/scg/lab_asbhatt/cookja/software/micromamba/envs/nanomotif_416/lib/python3.12/site-packages/nanomotif/dataload.py", line 47, in extract_contig_mods_with_sufficient_information
    contigmods_with_more_than_min_mods = pileup.groupby("contig_mod").count().filter(
                                         ^^^^^^^^^^^^^^
AttributeError: 'DataFrame' object has no attribute 'groupby'. Did you mean: 'group_by'?`

All the best,
Jane

**Comment by SorenHeidelbach** [comment]
We are working on fixing these issues

**Comment by SorenHeidelbach** [comment]
These error should have been fixed in 0.4.17, if they persist, do not hesitate to reopen this issue. #85
